### DISCUSSION TITLE
Feature Idea: Usage Resolution of Daily or Monthly, or an endpoint for accessing Billing data

### DISCUSSION THREAD

**Discussion by GrahamLea** [amberelectric/public-api#67] [Ideas]
I'm planning a new solar system and want to use our Amber usage data to build a model to predict our self-consumption and exports. I'm planning to pull our data from the last 12 months, if that's possible. 🤞🏼 I guess I'll probably aim eventually to get to an average daily kWh per month, similar to what's shown in the bill. That being the case, having the only `.../usage` resolution available be 30 minutes means I'm pulling way more data than I need. It'd be great to be able to ask for either daily or monthly for my use case. I guess there would be less fields coming back, as a bunch of them don't make sense at larger resolutions (e.g. `nemTime`), so maybe a different endpoint like `.../usage/daily` with a pared-back schema would be good.

Alternatively, having programmatic access to the data that goes into my emailed bill would probably be useful for my end goal.

**Comment by madpilot** [comment]
We were talking about this yesterday. As part of our 5-minute price work, we are completely reworking our usage storage, and part of that is the ability to do roll ups for reporting purposes. I haven't got specific timelines on when it'll be ready, but it's actively being worked on now, so will be the next big internal release.

That should fast follow with a billing engine that could possibly do your second suggestion.

TL;DR It's coming. Soon.

**Comment by madpilot** [comment]
BTW: Downloading a year of usage _should_ be fine. If you are having issues, let me know.

**Reply by GrahamLea** [reply]
Sweet. Thanks for that.
I shouldn't have trouble using the 30-minute data, but thought the daily roll-up would probably be a useful thing for future API users to have available.

**Reply by GrahamLea** [reply]
FYI: It doesn't work if I try to pull the whole year's worth in one request. One month at a time seems to work. I'm happy to do it in batches, but thought you might want to know in case you thought a request for a year of data should work.
```
amberelectric.exceptions.ServiceException: (502)
Reason: Bad Gateway
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'Content-Length': '36', 'Connection': 'keep-alive', 'Date': 'Sat, 11 Sep 2021 11:45:43 GMT', 'x-amzn-RequestId': '402b5c3f-02c8-459c-887f-6d24864fa8c8', 'x-amzn-ErrorType': 'InternalServerErrorException', 'x-amz-apigw-id': 'FfyBBESOSwMFfCA=', 'X-Amzn-Trace-Id': 'Root=1-613c96d3-61884dbf2c8b29ef76e54f3b', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 1f8f9cf6d7da51cc4ed1c2173cb0988c.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MEL50-C2', 'X-Amz-Cf-Id': 'Kv5fuuTaF5YqCDmfgMmA5eWOdjAN3anhAOMjPA7MUXzhzD15SnszbA=='})
HTTP response body: b'{"message": "Internal server error"}'
```

**Reply by madpilot** [reply]
The lambda would be timing out. The new usage engine will sort that out.

**Comment by GrahamLea** [comment]
In case anyone has the same need as me in the meantime, [I've made a tool that does this](https://github.com/amberelectric/public-api/discussions/70) and made it open source.
